### PR TITLE
Decoupling: Setup storybook playground for dashboard

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

Since the dashboard is now decoupled, there should be a storybook playground for the dashboard too.

## Summary

image

---

Explore Templates

image

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Related: https://github.com/google/web-stories-wp/pull/9569
Partially addresses https://github.com/google/web-stories-wp/issues/2918

### CODE REVIEW

**Comment by swissspidy** [comment]
Random thought: as a **stretch goal**, would it be possible for the dashboard playground to link to the editor playground when clicking on "Create New Story" for example?

Addons like https://storybook.js.org/addons/@storybook/addon-links might allow for this

**Comment by mohdsayed** [comment]
> Random thought: as a stretch goal, would it be possible for the dashboard playground to link to the editor playground when clicking on "Create New Story" for example?

@swissspidy Already doing it by passing the following, but without the use of the addon, and it works.

https://github.com/google/web-stories-wp/blob/c8a33866175a5a1ce56943582a57b2793888860e/.storybook/stories/playground/dashboard/index.js#L186

File: .storybook/stories/playground/story-editor/header/buttons/preview.js [hunk 1]
```diff
@@ -41,7 +46,9 @@ function PreviewButton() {
   const openPreviewLink = () => {
     saveStory();
 
-    const previewLink = `${window.location.origin}/iframe.html?id=playground-preview--default`;
+    const previewLink = getStorybookUrl(
+      'iframe.html?id=playground-preview--default'
+    );
```
Comment (swissspidy) [inline .storybook/stories/playground/story-editor/header/buttons/preview.js]: Something `@storybook/addon-links` could be used for?

File: packages/dashboard/src/utils/index.js [hunk 2]
```diff
@@ -23,3 +23,4 @@ export {
 } from './usePagePreviewSize';
 export { default as useStoryView } from './useStoryView';
 export { default as useTemplateView } from './useTemplateView';
+export { default as deepMerge } from './deepMerge';
```
Comment (swissspidy) [inline packages/dashboard/src/utils/index.js]: Maybe something for design system since it's used by story editor as well?

Comment (mohdsayed) [reply packages/dashboard/src/utils/index.js]: Ah okay, I thought you prefer to duplicate it until we decide something in https://github.com/google/web-stories-wp/issues/9639. Will move to design system

File: .storybook/stories/playground/story-editor/header/buttons/preview.js (hunk 1 above)
Comment (mohdsayed) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: Okay, trying...

File: packages/dashboard/src/utils/index.js (hunk 2 above)
Comment (swissspidy) [reply packages/dashboard/src/utils/index.js]: #9639 is for utils shared by the `wp-*` packages, no? Here we are in the core dashboard, not `wp-dashboard`

Comment (mohdsayed) [reply packages/dashboard/src/utils/index.js]: Yes, correct.

File: .storybook/stories/playground/story-editor/header/buttons/preview.js (hunk 1 above)
Comment (mohdsayed) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: It should be fine here for the preview, but the only problem is that we need to deal with the promise for updating `newStoryURL` in `config` using `hrefTo`, which is currently static and easy to pass. 

https://github.com/google/web-stories-wp/blob/c51869d083ff38a5f09240d18a7eb8898c6d742f/.storybook/stories/playground/dashboard/index.js#L126-L128

Comment (mohdsayed) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: For preview, it is easy though 

```js
const previewLink = await hrefTo('Playground/preview', 'default');
```

Comment (mohdsayed) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: But yeah `@storybook/addon-links` looks like a better choice, will think again.

Comment (swissspidy) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: It certainly feels less hardcoded and more future proof should storybook ever change the URL structure.

File: packages/dashboard/src/utils/index.js (hunk 2 above)
Comment (mohdsayed) [reply packages/dashboard/src/utils/index.js]: Moved in https://github.com/google/web-stories-wp/pull/9644/commits/8eb755628f1ca37180294374a63c9732ce6b1b5e

File: .storybook/stories/playground/story-editor/header/buttons/preview.js (hunk 1 above)
Comment (mohdsayed) [reply .storybook/stories/playground/story-editor/header/buttons/preview.js]: If we follow thier API we would need to call `linkTo` method using `onClick` for which we will have to make changes to the core dashboard.

So to keep things simple, I used the `hrefTo` from `@storybook/addon-links` for preview but  created a better utility function called  `linkHrefTo` using `toId` from `'@storybook/csf'` which is internally being used by `@storybook/addon-links`  to convert string to id.

Updated in https://github.com/google/web-stories-wp/pull/9644/commits/a79794fe060b7a3806d08bbd5338128be5f11d78#

**Review by swissspidy — APPROVED** [review]

**Pull request merged**